docs: explain the differences between activation and scaling values (#780)
kedacore/keda-docs: content/docs/2.8/concepts/scaling-deployments.md

The presensce of this annotation will pause autoscaling no matter what number of replicas is provided. The above annotation will scale your current workload to 0 replicas and pause autoscaling. You can set the value of replicas for an object to be paused at to any arbitary number. To enable autoscaling again, simply remove the annotation from the `ScaledObject`definition.

> ### Activating and Scaling thresholds

Reviewer: Would you mind updating all versions please?

Author: No no, this change applies only after next version because requires changes on every scaler

> ### Activating and Scaling thresholds

Reviewer: We should maybe use code formatting for configuration names

Reviewer: This comment was more for config names in this section, though. I think "Activating and Scaling thresholds" as a title as perfect

Reviewer: Can we rename it back?

Author: renamed it again to `Activating and Scaling thresholds`

> There are user cases where the activating value (0-1 and 1-0) is totally different than 0, ie: workloads scaled with prometheus scaler where the values go from -X to X. KEDA brings support to specify different values for each case:

Reviewer: Before we dive in to this, I would conceptually explain activation vs scaling, or point to our existing docs on this

Author: Isn't enough the explanation below? I can expand it

Reviewer: No because it only comes later on, so people who are new to this might have no idea what we are talking about

Reviewer 2: Agree with Tom on this one. We should start with explaining what is Activation and what is Scaling, then we can follow up with a different options.

> - Activation: This value will be used to consider the scaler active or not and scale from/to 0 based on it.
> - Scaling: This value will be passed to the HPA as the target value and the HPA controller will try to reach it.

Reviewer: What about this?
```suggestion
- **Activation:** Defines when the scaler is active or not and scales from/to 0 based on it.
- **Scaling:** Defines the target value to scale the workload from 1 to _n_ instances and vice versa. To achieve this, KEDA passes the target value to the Horizontal Pod Autoscaler (HPA) and the built-in HPA controller will handle all the autoscaling.
```

> ⚠️ **NOTE:** If the minimum replicas is >= 1, the scaler is always active and the Activation value will be ignored.

> Each scaler defines in its own docs the proper parameters for each case, but the key for specifying the activation value will be always the same as the scaling value, appending the prefix `activation` (ie: `threshold` for scaling and `activationThreshold` for activation).

Reviewer: What about this? However, not fully sure as this sentence is a bit confusing.
```suggestion
Each scaler defines parameters for their use-cases, but the activation can always be the same as the scaling value, appended by the prefix `activation` (ie: `threshold` for scaling and `activationThreshold` for activation).
```

The presensce of this annotation will pause autoscaling no matter what number of replicas is provided. The above annotation will scale your current workload to 0 replicas and pause autoscaling. You can set the value of replicas for an object to be paused at to any arbitary number. To enable autoscaling again, simply remove the annotation from the `ScaledObject`definition.

### `Activating` and `Scaling`

> There are use-cases where the activating value (0-1 and 1-0) is totally different than 0, such as workloads scaled with the Prometheus scaler where the values go from -X to X. To give a consisten solution to this problem, KEDA has 2 different phases during the autoscaling process.

Author: I think that done

> - **Activation phase:**: The activating (or deactivating) phase is the moment when KEDA (operator) has to decide if the workload should be scaled from/to zero. KEDA is the responsible for this action based on the result of the scaler `IsActive` function and only applies in from/to 0 scaling.

Author: done

> - **Scaling phase:** The scaling phase is the moment when KEDA has decided to scale out to 1 instance and now is the HPA controller who takes the scaling decisions based on the configuration defined in the generated HPA (from ScaledObject data) and the metrics exposed by KEDA (metrics server). This phase applies the scaling from 1-N and N-1.

Author: done

The presensce of this annotation will pause autoscaling no matter what number of replicas is provided. The above annotation will scale your current workload to 0 replicas and pause autoscaling. You can set the value of replicas for an object to be paused at to any arbitary number. To enable autoscaling again, simply remove the annotation from the `ScaledObject`definition.

### `Activating` and `Scaling`

There are use-cases where the activating value (0-1 and 1-0) is totally different than 0, such as workloads scaled with the Prometheus scaler where the values go from -X to X. To give a consisten solution to this problem, KEDA has 2 different phases during the autoscaling process.

- **Activation phase**
- **Scaling phase**

#### **Activating pahse**

The activating (or deactivating) phase is the moment when KEDA (operator) has to decide if the workload should be scaled from/to zero. KEDA is the responsible for this action based on the result of the scaler `IsActive` function and only applies in from/to 0 scaling.

#### **Scaling pahse**

> The scaling phase is the moment when KEDA has decided to scale out to 1 instance and now is the HPA controller who takes the scaling decisions based on the configuration defined in the generated HPA (from ScaledObject data) and the metrics exposed by KEDA (metrics server). This phase applies the scaling from 1-N and N-1.

Reviewer: Let's merge all of this in just the list to keep it simpler, maybe?

> #### **Thresholds**

Reviewer: No need for making it bold
```suggestion
#### Managing Activation & Scaling Thresholds
```

KEDA allows you to specify different values for each scenario:

> - **Activation:** Defines when the scaler is active or not and scales from/to 0 based on it.
> - **Scaling:** Defines the target value to scale the workload from 1 to _n_ instances and vice versa. To achieve this, KEDA passes the target value to the Horizontal Pod Autoscaler (HPA) and the built-in HPA controller will handle all the autoscaling.

Reviewer: We should use the configuration values here, for example:
```suggestion
- `activationThreshold`:  Defines when the scaler is active or not and scales from/to 0 based on it.
- `threshold`:  Defines the target value to scale the workload from 1 to _n_ instances and vice versa. To achieve this, KEDA passes the target value to the Horizontal Pod Autoscaler (HPA) and the built-in HPA controller will handle all the autoscaling.
```

Author: The problem is that we don't have specific values for that, they depend on the scaler, they can be queueLenght, threshold, value, lag... The constraint is that the activation value name will be the scaling value appending the prefix `activantion`, that's why I preferred to use these names

Author: But we can use `threshold` as the word for explaining the concept. WDYT?

Reviewer 2: yeah, let's keep the **Activation** and **Scaling** here

Author: No no, don't worry, this will be the base for the change so it's better to discuss it and take the best possible approach.
I prefer to make this clearest as possible before merging it and starting with these changes

Author: To clarify this, I have made it bold the paragraph where this is explained. I think that now it's clearer

The presensce of this annotation will pause autoscaling no matter what number of replicas is provided. The above annotation will scale your current workload to 0 replicas and pause autoscaling. You can set the value of replicas for an object to be paused at to any arbitary number. To enable autoscaling again, simply remove the annotation from the `ScaledObject`definition.

> ### `Activating` and `Scaling`

Reviewer: This is still not renamed as suggested here: https://github.com/kedacore/keda-docs/pull/780#discussion_r899774978

I think it would be better to revert to your original name given it's more approachable to new people and less technical focussed.

It also feels a bit odd to see this in conceptual docs:
![image](https://user-images.githubusercontent.com/4345663/175871063-4d550c61-b3e1-4391-b478-f775c189dbb6.png)

Author: I didn't see it, omw

Reviewer: No worries at all - That's what reviews are for - Thanks!

Author: reverted

To give a consisten solution to this problem, KEDA has 2 different phases during the autoscaling process.

> - **Activation phase:**: The activating (or deactivating) phase is the moment when KEDA (operator) has to decide if the workload should be scaled from/to zero. KEDA is the responsible for this action based on the result of the scaler`IsActive` function and only applies to 0<->1 scaling. There are use-cases where the activating value (0-1 and 1-0) is totally different than 0, such as workloads scaled with the Prometheus scaler where the values go from -X to X.

Reviewer: ```suggestion
- **Activation phase:** The activating (or deactivating) phase is the moment when KEDA (operator) has to decide if the workload should be scaled from/to zero. KEDA is the responsible for this action based on the result of the scaler`IsActive` function and only applies to 0<->1 scaling. There are use-cases where the activating value (0-1 and 1-0) is totally different than 0, such as workloads scaled with the Prometheus scaler where the values go from -X to X.
```

Author: updated

- **Scaling phase:** The scaling phase is the moment when KEDA has decided to scale out to 1 instance and now is the HPA controller who takes the scaling decisions based on the configuration defined in the generated HPA (from ScaledObject data) and the metrics exposed by KEDA (metrics server). This phase applies the to 1<->N scaling.

#### Managing Activation & Scaling Thresholds

KEDA allows you to specify different values for each scenario:

- **Activation:** Defines when the scaler is active or not and scales from/to 0 based on it.
- **Scaling:** Defines the target value to scale the workload from 1 to _n_ instances and vice versa. To achieve this, KEDA passes the target value to the Horizontal Pod Autoscaler (HPA) and the built-in HPA controller will handle all the autoscaling.

> ⚠️ **NOTE:** If the minimum replicas is >= 1, the scaler is always active and the activation value will be ignored.

> **Each scaler defines parameters for their use-cases, but the activation will always be the same as the scaling value, appended by the prefix `activation`** (ie: `threshold` for scaling and `activationThreshold` for activation).

Reviewer: I don't think we need to put this in bold as it looks odd in the rendered docs IMO; but who am I!
```suggestion
Each scaler defines parameters for their use-cases, but the activation will always be the same as the scaling value, appended by the prefix `activation` (ie: `threshold` for scaling and `activationThreshold` for activation).
```

Author: addressed
